newIDE/app/src/ObjectEditor/ObjectEditorDialog.js

```diff
@@ -85,13 +85,11 @@ const InnerDialog = (props: InnerDialogProps) => {
     onCancel: props.onCancel,
   });
 
-  const objectMetadata = React.useMemo(
-    () =>
-      gd.MetadataProvider.getObjectMetadata(
-        props.project.getCurrentPlatform(),
-        props.object.getType()
-      ),
-    [props.project, props.object]
+  // Don't use a memo for this because metadata instances are built from
+  // scratch when extensions are refreshed.
+  const objectMetadata = gd.MetadataProvider.getObjectMetadata(
```

Comment:
Nice bug

Reply:
When was this causing a problem?

Reply:
When attaching a behavior that needs an extension installation on a custom object.
As the metadata is generated from the EBO, the old one is dead.